test/functional/testlib.bash

```diff
@@ -0,0 +1,502 @@
+#!bin/bash
+
+FUNC_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+TEST_FILENAME=$(basename "$BATS_TEST_FILENAME")
+TEST_NAME=${TEST_FILENAME%.bats}
+export TEST_NAME
+export FUNC_DIR
+export SWUPD_DIR="$FUNC_DIR/../.."
+export SWUPD="$SWUPD_DIR/swupd"
+
+
+generate_random_content() { 
+
+	local bottom_range=5
+	local top_range=100
+	local range=$((top_range - bottom_range + 1))
+	local number_of_lines=$((RANDOM%range))  
+	< /dev/urandom tr -dc 'a-zA-Z0-9-_!@#$%^&*()_+{}|:<>?=' | fold -w 100 | head -n $number_of_lines
```

Comment:
Really? It is very straight forward. read bytes from /dev/urandom, delete any that are not listed, pack them into lines a hundred long and get the first N of them.

What is concerning is the bug in the previous line, where @castulo failed to add bottom_range into number_of_lines, so this will generate between 0 and 95 lines (inclusive) of output, rather than between 5 and 100.

Reply:
you are right, I forgot to add bottom_range, thanks for the comment